### ISSUE TITLE
Poco::format() doesn't support (space) as a flag like printf() does

### ISSUE THREAD

**Issue by austin-beer** [pocoproject/poco#2730]
printf() supports a space ' ' as a flag in the format specifier. It would be great if Poco::format() supported the same.

http://www.cplusplus.com/reference/cstdio/printf/
https://github.com/pocoproject/poco/blob/develop/Foundation/src/Format.cpp

**Comment by austin-beer** [comment]
This issue has been open for over two and a half years. Is there any chance of addressing this issue?

**Comment by austin-beer** [comment]
This issue has been open for over **three** years? Would it be possible for someone to address this issue?

**Comment by austin-beer** [comment]
Yes, this is still an issue in Poco 1.13. This switch statement still doesn't handle a space character in the format string: https://github.com/pocoproject/poco/blob/poco-1.13.0-release/Foundation/src/Format.cpp#L35

Unfortunately I don't have time right now to develop a fix for this, but I'll put it on my TODO list to work on if I get time.

**Comment by austin-beer** [comment]
Ah, yes, I didn't realize that. It does make implementing this capability more difficult. Perhaps instead then an exception should be thrown if ' ' is encountered as a format flag so that the un-expected behavior (that it's not supported) is clear to anyone who tries to use it.

**Comment by austin-beer** [comment]
Any further thoughts from anyone on throwing an exception if a space is encountered as a format flag in calls to `Poco::format()`, since it's not supported?

**Comment by aleks-f** [comment]
maybe do the same thing when argument does not match format placeholder:

https://github.com/pocoproject/poco/blob/4c2128ccdb3bd9968778b7b76ed9641f79ccd16c/Foundation/include/Poco/Format.h#L98-L100
